### PR TITLE
Add fulltexts and autoresolve

### PR DESCRIPTION
**Pull request by maboelnour**
Add docs for fulltexts, and autoresolve endpoints.

### CODE REVIEW

File: .insomnia/Request/req_269890eb42284f299fa166e939a6b566.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  A succesful request returns the created fulltext as JSON.
+method: POST
+body:
+  mimeType: application/json
+  text: >-
+    {
+    	"upload": {
+    		"s3_key": "{% response 'body', 'req_3a5cf8c800c241e1adc5e379037c9417', 'b64::L1Bvc3RSZXNwb25zZS9LZXk=::46b', 'never', 60 %}",
+    		"original_filename": "search_file.ris",
```
Comment (hammady) [inline .insomnia/Request/req_269890eb42284f299fa166e939a6b566.yml:29]: This should be fulltext PDF.

File: .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      value: "{% response 'body', 'req_8ed2f5bc95e24396af82fbc0420b79d2',
+        'b64::JC5maWVsZHMuc3VjY2Vzc19hY3Rpb25fc3RhdHVz::46b', 'never', 60 %}"
+      description: ""
+    - id: pair_ee50fadd0522413eb4891b4680f946f7
+      name: file
+      value: ""
+      description: ""
+      type: file
+      multiline: false
+      fileName: C:\Users\maboelnour\Downloads\gibson.jpg
```
Comment (hammady) [inline .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml:55]: Should be empty.

File: .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+url: "{{ _.base_url }}/api/v1/reviews/{{ _.review_id }}/autoresolve"
+name: Auto Resolve
+description: >-
+  This endpoint triggerrs an `autoresolve` job for the given review
+
+
+
+  Params:
+
+  - `autoresolve_percentage` (int) value should be between `0` and `99`
```
Comment (hammady) [inline .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml:15]: The backend doesn't validate this value. I found that 100 is also acceptable and has a valid use-case. I suggest the below change:
```suggestion
  - `autoresolve_percentage` (int) value should be between `0` and `100`, inclusive.
```

File: .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+_id: req_8f843e785e1c442d961a9ed0437d289c
+type: Request
+parentId: fld_80bc7ee1b14547dc860545b5af4878c5
+modified: 1681910084632
+created: 1681907101878
+url: "{{ _.base_url }}/api/v1/reviews/{{ _.review_id }}/autoresolve"
+name: Auto Resolve
+description: >-
+  This endpoint triggerrs an `autoresolve` job for the given review
```
Comment (hammady) [inline .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml:9]: We can mention here that once the job is done, a Pusher event will be sent out to subscribers.

File: .insomnia/Request/req_97704eb8c63c4555b2f939314975951a.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+_id: req_97704eb8c63c4555b2f939314975951a
+type: Request
+parentId: fld_4a309c10b7ae4963b8f7c5570d6f23fe
+modified: 1681965060989
+created: 1681964981137
+url: "{{ _.base_url }}/api/v1/fulltexts/{% response 'body',
+  'req_269890eb42284f299fa166e939a6b566', 'b64::JC5pZA==::46b', 'never', 60 %}"
+name: Fulltext
+description: ""
```
Comment (hammady) [inline .insomnia/Request/req_97704eb8c63c4555b2f939314975951a.yml:9]: Missing.

File: .insomnia/Request/req_d7a05c6d1d4f4bebaa56c46d7d91bb7a.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+_id: req_d7a05c6d1d4f4bebaa56c46d7d91bb7a
+type: Request
+parentId: fld_4a309c10b7ae4963b8f7c5570d6f23fe
+modified: 1681965095693
+created: 1681965067585
+url: "{{ _.base_url }}/api/v1/fulltexts/{% response 'body',
+  'req_269890eb42284f299fa166e939a6b566', 'b64::JC5pZA==::46b', 'never', 60 %}"
+name: Delete Fulltext
+description: ""
```
Comment (hammady) [inline .insomnia/Request/req_d7a05c6d1d4f4bebaa56c46d7d91bb7a.yml:9]: Missing.

File: .insomnia/Request/req_e5e3b9473a604f48af0c2936570eaacc.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+_id: req_e5e3b9473a604f48af0c2936570eaacc
+type: Request
+parentId: fld_4a309c10b7ae4963b8f7c5570d6f23fe
+modified: 1681964984542
+created: 1681920995044
+url: "{{ _.base_url }}/api/v1/fulltexts/"
+name: Fulltexts
+description: ""
```
Comment (hammady) [inline .insomnia/Request/req_e5e3b9473a604f48af0c2936570eaacc.yml:8]: Missing.

File: .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
+_id: req_3a5cf8c800c241e1adc5e379037c9417
+type: Request
+parentId: fld_4a309c10b7ae4963b8f7c5570d6f23fe
+modified: 1681964157694
+created: 1681921383211
+url: "{% response 'body', 'req_8ed2f5bc95e24396af82fbc0420b79d2',
+  'b64::JC51cmw=::46b', 'never', 60 %}"
+name: Upload Fulltext File to S3
+description: ""
```
Comment (hammady) [inline .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml:9]: Missing.

**Review by hammady — CHANGES_REQUESTED** [review]

File: .insomnia/Request/req_269890eb42284f299fa166e939a6b566.yml (hunk 1 above)
Comment (maboelnour) [reply .insomnia/Request/req_269890eb42284f299fa166e939a6b566.yml:29]: Done.

File: .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml (hunk 2 above)
Comment (maboelnour) [reply .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml:55]: Done.

File: .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml (hunk 4 above)
Comment (maboelnour) [reply .insomnia/Request/req_8f843e785e1c442d961a9ed0437d289c.yml:9]: Done.

File: .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml (hunk 8 above)
Comment (maboelnour) [reply .insomnia/Request/req_3a5cf8c800c241e1adc5e379037c9417.yml:9]: Done.

**Review by hammady — CHANGES_REQUESTED** [review]
There are still a few missing descriptions.

File: .insomnia/Request/req_e5e3b9473a604f48af0c2936570eaacc.yml (hunk 7 above)
Comment (maboelnour) [reply .insomnia/Request/req_e5e3b9473a604f48af0c2936570eaacc.yml]: Added.

File: .insomnia/Request/req_d7a05c6d1d4f4bebaa56c46d7d91bb7a.yml (hunk 6 above)
Comment (maboelnour) [reply .insomnia/Request/req_d7a05c6d1d4f4bebaa56c46d7d91bb7a.yml]: Added.

File: .insomnia/Request/req_97704eb8c63c4555b2f939314975951a.yml (hunk 5 above)
Comment (maboelnour) [reply .insomnia/Request/req_97704eb8c63c4555b2f939314975951a.yml]: Added.

**Pull request merged**